hoot-core/src/main/cpp/hoot/core/io/OsmApiChangeset.cpp

```diff
@@ -417,18 +430,39 @@ bool XmlChangeset::addNode(ChangesetInfoPtr& changeset, ChangesetType type, XmlN
   //  Only add the nodes that are "sendable"
   if (canSend(node))
   {
-    //  Add create nodes if the ID map's ID is negative, modify and delete IDs don't matter
+    //  Add create nodes if the ID map's ID is negative, modify IDs don't matter
     if ((type == ChangesetType::TypeCreate && _idMap.getNewId(ElementType::Node, node->id()) < 0) ||
-         type != ChangesetType::TypeCreate)
+         type != ChangesetType::TypeDelete)
     {
       //  Add the node
       changeset->add(ElementType::Node, type, node->id());
       markBuffered(node);
+      return true;
+    }
+    else if (type == ChangesetType::TypeDelete)
```

Comment:
Delete operations now have to check parents before adding them to the changeset.